lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -106,10 +107,9 @@ def on_class_node_enter(node)
       entry = Entry::Class.new(
         nesting,
         @file_path,
-        node.location,
-        constant_path.location,
+        Location.from_prism_location(node.location, @encoding),
```

Comment:
What do you think about adding factory methods to the `Entry` classes instead? e.g.:

```ruby
      entry = Entry::Class.from_prism_location(
        nesting,
        @file_path,
        node,
        constant_path,
        @encoding,
        comments,
        parent_class,
      )
```
(as well as `from_location` for when we're dealing with `Location`).

Reply:
That's going to preserve the coupling that I'm trying to get rid of. We need to be able to build entries without depending on the encoding, so that in certain situations we instantiate `RubyIndexer::Location` outside without passing the encoding around.

Reply:
Ok, let's go with this for now, but I think there's probably a way to clean up the API later.